server/datastore/mysql/policies_test.go

```diff
@@ -941,83 +1045,89 @@ func testTeamPolicyTransfer(t *testing.T, ds *Datastore) {
+	host2, err := ds.EnrollHost(ctx, "2", "2", &team1.ID, 0)
 	require.NoError(t, err)
 
-	require.NoError(t, ds.AddHostsToTeam(context.Background(), &team1.ID, []uint{host1.ID}))
-	host1, err = ds.Host(context.Background(), host1.ID)
+	require.NoError(t, ds.AddHostsToTeam(ctx, &team1.ID, []uint{host1.ID}))
+	host1, err = ds.Host(ctx, host1.ID)
 	require.NoError(t, err)
 
-	tq, err := ds.NewQuery(context.Background(), &fleet.Query{
+	tq, err := ds.NewQuery(ctx, &fleet.Query{
 		Name:        "query1",
 		Description: "query1 desc",
 		Query:       "select 1;",
 		Saved:       true,
 	})
 	require.NoError(t, err)
-	teamPolicy, err := ds.NewTeamPolicy(context.Background(), team1.ID, &user1.ID, fleet.PolicyPayload{
+	team1Policy, err := ds.NewTeamPolicy(ctx, team1.ID, &user1.ID, fleet.PolicyPayload{
 		QueryID: &tq.ID,
 	})
 	require.NoError(t, err)
 
-	gq, err := ds.NewQuery(context.Background(), &fleet.Query{
+	gq, err := ds.NewQuery(ctx, &fleet.Query{
 		Name:        "query2",
 		Description: "query2 desc",
 		Query:       "select 2;",
 		Saved:       true,
 	})
 	require.NoError(t, err)
-	globalPolicy, err := ds.NewGlobalPolicy(context.Background(), &user1.ID, fleet.PolicyPayload{
+	globalPolicy, err := ds.NewGlobalPolicy(ctx, &user1.ID, fleet.PolicyPayload{
 		QueryID: &gq.ID,
 	})
 	require.NoError(t, err)
 
-	require.NoError(t, ds.RecordPolicyQueryExecutions(context.Background(), host1, map[uint]*bool{teamPolicy.ID: ptr.Bool(false), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
-	require.NoError(t, ds.RecordPolicyQueryExecutions(context.Background(), host1, map[uint]*bool{teamPolicy.ID: ptr.Bool(true), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
-	require.NoError(t, ds.RecordPolicyQueryExecutions(context.Background(), host2, map[uint]*bool{teamPolicy.ID: ptr.Bool(false), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
-	require.NoError(t, ds.RecordPolicyQueryExecutions(context.Background(), host2, map[uint]*bool{teamPolicy.ID: ptr.Bool(true), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
+	require.NoError(t, ds.RecordPolicyQueryExecutions(ctx, host1, map[uint]*bool{team1Policy.ID: ptr.Bool(false), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
+	require.NoError(t, ds.RecordPolicyQueryExecutions(ctx, host1, map[uint]*bool{team1Policy.ID: ptr.Bool(true), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
+	require.NoError(t, ds.RecordPolicyQueryExecutions(ctx, host2, map[uint]*bool{team1Policy.ID: ptr.Bool(false), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
+	require.NoError(t, ds.RecordPolicyQueryExecutions(ctx, host2, map[uint]*bool{team1Policy.ID: ptr.Bool(true), globalPolicy.ID: ptr.Bool(true)}, time.Now(), false))
 
-	checkPassingCount := func(expectedCount, expectedGlobalCount uint) {
-		policies, err := ds.ListTeamPolicies(context.Background(), team1.ID)
+	checkPassingCount := func(tm1, tm1Inherited, tm2Inherited, global uint) {
```

Comment:
nothing to do with this PR, just a general comment: this helper function is so confusing to me, every time I have to look at it I spend a good time figuring out what it does.

Reply:
I agree, I kept it as I was only altering the assertions and didn't want to change the whole test around, but it is a bit confusing.